tests/std/tests/P0896R4_ranges_alg_uninitialized_copy/test.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <memory>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate that uninitialized_copy_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::uninitialized_copy_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_copy(borrowed<false>{}, borrowed<false>{})),
+    ranges::uninitialized_copy_result<ranges::dangling, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_copy(borrowed<false>{}, borrowed<true>{})),
+    ranges::uninitialized_copy_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_copy(borrowed<true>{}, borrowed<false>{})),
+    ranges::uninitialized_copy_result<int*, ranges::dangling>>);
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_copy(borrowed<true>{}, borrowed<true>{})),
+    ranges::uninitialized_copy_result<int*, int*>>);
+
+struct int_wrapper {
+    inline static int constructions = 0;
+    inline static int destructions  = 0;
+
+    static void clear_counts() {
+        constructions = 0;
+        destructions  = 0;
+    }
+
+    static constexpr int magic_throwing_val = 29;
+    int val                                 = 10;
+
+    int_wrapper() {
+        ++constructions;
+    }
+    int_wrapper(int x) : val{x} {
+        ++constructions;
+    }
+
+    int_wrapper(const int_wrapper& that) {
+        if (that.val == magic_throwing_val) {
+            throw magic_throwing_val;
+        }
+
+        val = that.val;
+        ++constructions;
+    }
+
+    ~int_wrapper() {
+        ++destructions;
+    }
+
+    int_wrapper& operator=(const int_wrapper&) {
+        abort();

```

Comment:
Should include `<cstdlib>` for `abort()`. Occurs in the other tests.

Reply:
Done